docs/src/variables.md

```diff
@@ -470,7 +510,28 @@ Dict{Symbol,Symmetric{JuMP.VariableRef,Array{JuMP.VariableRef,2}}} with 2 entrie
 
 ## Deleting variables
 
-**TODO(@odow):** explain how to delete variables.
+Finally, JuMP supports the deletion of optimization variables. However, before
+we explain how to do so, it is useful to add a short alias for
+`MathOptInterface`:
+```jldoctest variables_delete; setup=:(model=Model())
+julia> const MOI = JuMP.MathOptInterface
+MathOptInterface
+```
+
+To delete variables, we can use the `MOI.delete!` method. We can also check
```

Comment:
Let's implement `JuMP.delete!` and `JuMP.is_valid` instead of documenting the MOI way.